v2/internal/controllers/controller_resources.go

```diff
@@ -338,12 +336,6 @@ func watchEntity(c client.Client, log logr.Logger, keys []string, objList client
 			return nil
 		}
 
-		// This should be fast since the list of items it's going through should always be cached
-		// locally with the shared informer.
-		// Unfortunately we don't have a ctx we can use here, see https://github.com/kubernetes-sigs/controller-runtime/issues/1628
-		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Do we still need to have a 10s timeout, or is this fine?

Reply:
I brought this back.